File: Coding/Java/Reverse_String.java
public class ReverseString {

    public static void main(String[] args) {
        String name = "Nikhil";
        
        // 1. Using StringBuilder
        System.out.println(new StringBuilder(name).reverse().toString());
        
        // 2. Reverse using one pointer
        String revName = "";
        for (int i = name.length() - 1; i >= 0; i--) {
            revName += name.charAt(i);
        }
        System.out.println(revName);

        // 3. Two pointers (in-place reverse)
        char[] nameArray = name.toCharArray();
        int left = 0;
        int right = nameArray.length - 1;

        while (left < right) {
            char temp = nameArray[left];
            nameArray[left] = nameArray[right];
            nameArray[right] = temp;
            left++;
            right--;
        }

        String reversedInPlace = new String(nameArray);
        System.out.println("Reversed with two pointers (in-place): " + reversedInPlace);
    }
}

File: Coding/Python/Reverse_String.py
name = "Nikhil"

# Slicing
print(name[::-1])

# One Pointer
rev_name = ""
for i in range(len(name) - 1, -1, -1):
    rev_name += name[i]
print(rev_name)

# Two Pointer (in-place reverse)
name_list = list(name)
left = 0
right = len(name_list) - 1

while left < right:
    name_list[left], name_list[right] = name_list[right], name_list[left]
    left += 1
    right -= 1

inplace_name = "".join(name_list)
print(inplace_name)